awx/api/views/__init__.py

```diff
@@ -480,7 +479,6 @@ class InstanceGroupDetail(RelatedJobsPreventDeleteMixin, RetrieveUpdateDestroyAP
     name = _("Instance Group Detail")
     model = models.InstanceGroup
     serializer_class = serializers.InstanceGroupSerializer
-    permission_classes = (InstanceGroupTowerPermission,)
```

Comment:
I believe this was intended to case a 405 error with a DELETE request. Have you tested that a DELETE request will still give some 400-level error after this change? Because I don't see how. If you don't use the permissions class (which I think is fine... personally), then you probably need to handle it in the view, probably via the `perform_destroy` method.

Reply:
With the current change we get a 403.

```HTTP/1.1 403 Forbidden
Server: nginx
Date: Wed, 02 Mar 2022 19:52:44 GMT
Content-Type: application/json
Content-Length: 63
Connection: keep-alive
Vary: Accept, Accept-Language, Origin, Cookie
Allow: GET, PUT, PATCH, DELETE, HEAD, OPTIONS
X-API-Product-Version: 20.0.1.dev35+gf65fe0de86
X-API-Product-Name: AWX
X-API-Node: awx_1
X-API-Time: 0.113s
X-API-Query-Count: 3
X-API-Query-Time: 0.003s
Content-Language: en
X-API-Total-Time: 0.174s
X-API-Request-Id: 98dfbeea1de44a8db050893b2b2b8da2
Access-Control-Expose-Headers: X-API-Request-Id
```

``````bash
{"detail":"You do not have permission to perform this action."}
````

Reply:
Okay, as long as it doesn't actually delete, I have no preference, myself

Reply:
I think 403 is preferable in this circumstance, where the decision to allow delete is dynamic.  405 strikes me as meaning that the endpoint doesn't support the verb in general.